### ISSUE TITLE
With Latest Android Intercom SDK 10.0.0, Handling the FCM background message not working. The app freezes.

### ISSUE THREAD

**Issue by deepak786** [deepak786/intercom_flutter#151]
```
/// top level function to listen for messages in the background
Future<void> _firebaseMessagingBackgroundHandler(RemoteMessage message) async {
  print('Handling a background message ${message.messageId}');
  // If you're going to use other Firebase services in the background, such as Firestore,
  // make sure you call `initializeApp` before using other Firebase services.
  await Firebase.initializeApp();

    // check if the push message is from Intercom
    bool intercomSupported = await isIntercomSupported();
    if (intercomSupported) {
      bool isIntercomPush = await Intercom.isIntercomPush(data);
      if (isIntercomPush) {
        await Intercom.handlePush(data);
        return;
      }
    }

   /// other code
}
```

This is how I'm handling the background push notification from FCM. Here I'm checking if the push is Intercom push or not. It was working fine with Intercom SDK 8.2.+. But as the Android Intercom SDK is updated to 10.0.0, it is not working, the app just got hanged.

using the below version of Intercom SDK:

```
intercom_flutter:
    git:
      url: https://github.com/deepak786/intercom_flutter
      path: intercom_flutter
      ref: ios_send_token_patch
```

@obrunsmann @VladimirMitso could you guys please check that if you guys are also facing the same issue. As you guys have already tested the patch for iOS. Could you please test the same for Android?

Logs:
```
V/FA      (10368): Inactivity, disconnecting from the service
D/PushInterceptReceiver(10368): Intercom message received
D/CompatibilityChangeReporter(10368): Compat change id reported: 160794467; UID 10159; state: ENABLED
I/endaboa.app.de(10368): Thread[5,tid=10374,WaitingInMainSignalCatcherLoop,Thread*=0x7d353ec319a0,peer=0x12e401f8,"Signal Catcher"]: reacting to signal 3
I/endaboa.app.de(10368): 
I/endaboa.app.de(10368): Wrote stack traces to tombstoned
Lost connection to device.
```

Seems that someone already reported the issue to Intercom https://forum.intercom.com/s/question/0D52G000052LVfRSAW/android-push-notification-not-working

**Comment by deepak786** [comment]
I have created a patch to downgrade the Intercom Android SDK version to `8.3.0`
```
intercom_flutter:
    git:
      url: https://github.com/deepak786/intercom_flutter
      path: intercom_flutter
      ref: android_intercom_downgrade_patch
```

With SDK version `8.3.0`, it is working fine.

**Comment by deepak786** [comment]
I tested with the latest Intercom Android SDK version `10.4.0`. Still, it is not working.

The below method is not working when the app is in the background.
```
Intercom.handlePush(message);
```

**Comment by mabdurrahman** [comment]
@deepak786 I did a bit more deep investigation yesterday and I was able to identify the exact Intercom Android SDK version where our issue started, it's v.9.2.0, so the Android notifications are working fine on/pre v.9.1.2.

I posted the same finding to Intercom support hoping it helps them move this quickly.

**Comment by bbilly-git** [comment]
Hi @deepak786 @mabdurrahman I have the same issue, hopefully Intercom can fix this soon

**Comment by akilarajeshks** [comment]
Looks like this has been fixed in v10.6.1 that was released today.
